api/src/main/java/com/codeforcommunity/exceptions/LeaderCannotLeaveTeamException.java

```diff
@@ -0,0 +1,27 @@
+package com.codeforcommunity.exceptions;
+
+import com.codeforcommunity.rest.FailureHandler;
+import io.vertx.ext.web.RoutingContext;
+
+public class LeaderCannotLeaveTeamException extends HandledException {
```

Comment:
To clarify this, as I commented on the dev docs PR, can a leader leave if there is more than one leader?

Reply:
No. It would be fine for a leader to leave a team if there is more than one leader. However, there is currently no route for making someone else a team leader without transferring ownership. When a team is created there is only one leader. Given the current API Spec there is no way to get into a state with more than one leader. 
Might want to add this in the future. Would not be a complicated fix.